core/state_processor.go

```diff
@@ -150,12 +155,25 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 					}
 
 					if acc.HasStorage() {
+					stIt, err := statedb.Snaps().StorageIterator(mpt.Hash(), accIt.Hash(), common.Hash{})
+					if err != nil {
+						return nil, nil, 0, err
+					}
 						for ; stIt.Next() && count < maxMovedCount; count++ {
+						var (
+							value     []byte   // slot value after RLP decoding
+							safeValue [32]byte // 32-byte aligned value
+						)
+						if err := rlp.DecodeBytes(stIt.Slot(), &value); err != nil {
+							return nil, nil, 0, fmt.Errorf("error decoding bytes %x: %w", stIt.Slot(), err)
+						}
+						copy(safeValue[32-len(value):], value)
 							slotnr := rawdb.ReadPreimage(statedb.Database().DiskDB(), stIt.Hash())
+						fdb.LastSlotHash = stIt.Hash()
 
-							mkv.addStorageSlot(addr, slotnr, stIt.Slot())
+						mkv.addStorageSlot(addr, slotnr, safeValue[:])
 						}
-					}
+					stIt.Release()
```

Comment:
For EOA addresses that don't enter the `if` in line L157, we never call `stIt.Release()`. Should we?

Honestly, this `Release()` calls for account and storage iterators start to feel a bit tricky; maybe with some method encapsulation we can find a way to always handle them with `defer` somehow? We have to be careful here into consider all branches/early-returns and remember to release.